src/app_service/service/community/service.nim

```diff
@@ -590,47 +587,61 @@ QtObject:
 
   proc init*(self: Service) =
     self.doConnect()
-    self.communityTags = self.loadCommunityTags();
 
-    let communities = self.loadAllCommunities()
-    for community in communities:
-      self.communities[community.id] = community
-      if (community.admin):
-        self.communities[community.id].pendingRequestsToJoin = self.pendingRequestsToJoinForCommunity(community.id)
-        self.communities[community.id].declinedRequestsToJoin = self.declinedRequestsToJoinForCommunity(community.id)
-        self.communities[community.id].canceledRequestsToJoin = self.canceledRequestsToJoinForCommunity(community.id)
+    try:
+      let arg = AsyncLoadCommunitiesDataTaskArg(
+        tptr: cast[ByteAddress](asyncLoadCommunitiesDataTask),
+        vptr: cast[ByteAddress](self.vptr),
+        slot: "asyncCommunitiesDataLoaded",
+      )
+      self.threadpool.start(arg)
+    except Exception as e:
+      error "Error requesting communities data", msg = e.msg
 
-    let communitiesSettings = self.loadCommunitiesSettings()
-    for settings in communitiesSettings:
-      if self.communities.hasKey(settings.id):
-        self.communities[settings.id].settings = settings
+  proc asyncCommunitiesDataLoaded(self: Service, response: string) {.slot.} =
+    try:
+      let responseObj = response.parseJson
+      if (responseObj{"error"}.kind != JNull and responseObj{"error"}.getStr != ""):
+        error "error loading communities data", msg = responseObj{"error"}.getStr
```

Comment:
1. Should we do a return here?
2. What should happen, if we'll get an error? Do we already handle somewhere the case, when loading is failed? Do we need additional event for this?

Reply:
Fixed